### ISSUE TITLE
Operates slow in folder with a lot of files

### ISSUE THREAD

**Issue by foegra** [lynchjames/obsidian-writing-goals#51]
I do have a journal folder with files for every day starting from a couple of years ago. Setting a goal for the whole folder - makes typing process freeze and lagging.

**Comment by Azmoinal** [comment]
I noticed also on each device I have. 

The lag is really noticeable. After some time, I had to remove folder goals relying only on note goals, while using novel word count for folder goal.

Unfortunately novel word count doesn't have a daily goal but only the total count.

I hope it could be fixed.

**Comment by Medullitus** [comment]
Is the developer looking here?? I'm getting a 'FAILED TO SHOW GOAL' error!!

**Comment by Medullitus** [comment]
UPDATE: I found the issue! The HOMEPAGE plugin

When the Homepage plugin is enabled, the daily word goal plugin doesn't open and gives an error. You can disable the Homepage plugin, open the word goal window, and then reactivate the Homepage plugin. Thanks. @lynchjames  we love you and your plugin is amazing! Please don't leave us!! Thanks <3<3<3<3<3

**Comment by lynchjames** [comment]
I've made a change to updating goal counts every few seconds rather than on each file update which was too expensive for larger vaults. There will be a setting to allow you to change the time in seconds between goal updates.

I have also added more caching to reduce the amount of note processing required.

With these improvements in place, I can see noticeable improvements in performance on a vault with over 800 notes.
